tools/test_wire_format_reference.py

```
@@ -0,0 +1,63 @@
+    # The regression itself: both commands must refuse, not just `verify`.
+    ("verify under -O", ["-O"], ["verify"], 1),
+    ("generate under -O", ["-O"], ["generate"], 1),
+    # -OO strips docstrings as well as asserts; same refusal must hold.
+    ("verify under -OO", ["-OO"], ["verify"], 1),
+]
+
+
+def main() -> int:
+    failures = []
+    for name, flags, argv, expected in CASES:
+        proc = subprocess.run(
```

Ruff (0.16.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)